tests/codex/testvalidation.nim

```diff
@@ -41,12 +56,170 @@ asyncchecksuite "validation":
   test "the list of slots that it's monitoring is empty initially":
     check validation.slots.len == 0
 
+  for (validationGroups, groupIndex) in [(100, 100'u16), (100, 101'u16)]:
+    test "initializing ValidationConfig fails when groupIndex is " &
+        "greater than or equal to validationGroups " &
+        fmt"(testing for {groupIndex = }, {validationGroups = })":
+      let groups = ValidationGroups(validationGroups).some
+      let validationConfig = ValidationConfig.init(
+          maxSlots, groups = groups, groupIndex = groupIndex)
+      check validationConfig.isFailure == true
+      check validationConfig.error.msg == "The value of the group index " &
+          "must be less than validation groups! " &
+          fmt"(got: {groupIndex = }, groups = {!groups})"
+  
+  test "initializing ValidationConfig fails when maxSlots is negative":
+    let maxSlots = -1
+    let validationConfig = ValidationConfig.init(
+        maxSlots = maxSlots, groups = ValidationGroups.none)
+    check validationConfig.isFailure == true
+    check validationConfig.error.msg == "The value of maxSlots must " &
+        fmt"be greater than or equal to 0! (got: {maxSlots})"
+  
+  test "initializing ValidationConfig fails when maxSlots is negative " &
+      "(validationGroups set)":
+    let maxSlots = -1
+    let validationConfig = ValidationConfig.init(
+        maxSlots = maxSlots, groups = validationGroups, groupIndex)
+    check validationConfig.isFailure == true
+    check validationConfig.error.msg == "The value of maxSlots must " &
+        fmt"be greater than or equal to 0! (got: {maxSlots})"
+
+  test "group index is irrelevant if validation groups are not set":
+    randomize()
+    for _ in 0..<100:
+      let groupIndex = rand(1000).uint16
+      let validationConfig = ValidationConfig.init(
+          maxSlots, groups=ValidationGroups.none, groupIndex)
+      check validationConfig.isSuccess
+  
+  test "slot should be observed if it is in the validation group":
+    let validationConfig = initValidationConfig(
+        maxSlots, validationGroups, groupIndex)
+    let validation = Validation.new(clock, market, validationConfig)
+    check validation.shouldValidateSlot(slot.id) == true
+  
+  test "slot should be observed if validation group is not set":
+    let validationConfig = initValidationConfig(
+        maxSlots, ValidationGroups.none)
+    let validation = Validation.new(clock, market, validationConfig)
+    check validation.shouldValidateSlot(slot.id) == true
+  
+  test "slot should be observed if maxSlots is set to 0":
+    let validationConfig = initValidationConfig(
+        maxSlots = 0, ValidationGroups.none)
+    let validation = Validation.new(clock, market, validationConfig)
+    check validation.maxSlotsConstraintRespected
```

Comment:
I would expect this test to call `shouldValidateSlot` at the end. Also, imo, `maxSlotsConstraintRespected` shouldn't be exported from the Validation module as it's more an internal logic routine.

Reply:
This is one of those tests that I would not mind removing in the first place. I added them when I did not have an idea on how to test this properly. Those tests are testing internals - normally I do not like to add such tests, but sometimes I do when I do not have better idea... Thus, if you do not mind I would remove that one...

Reply:
I think it's still a valid test, as it checks a different branch in the logic. Perhaps the last line could be changed to `check validation.shouldValidateSlot(slot.id) == true`? Wdyt?

Reply:
Will be covered by https://github.com/codex-storage/nim-codex/pull/890#discussion_r1772975534.